app/models/membership.rb

```diff
@@ -1,9 +1,16 @@
 # encoding: utf-8
 
 class Membership < BaseModel
-
   belongs_to :user, -> { with_deleted }, touch: true
   belongs_to :account, -> { with_deleted }, touch: true
 
   scope :approved, -> { where(approved: true) }
+
+  after_commit :update_user_default_account, on: [:destroy]
+
+  def update_user_default_account
+    if user.default_account == account
+      user.update_attributes!(account_id: user.individual_account.id)
```

Comment:
yes. the logic is taken from V3. If you lose membership to the account that is currently your default, we update the default to your individual account rather than having it continue to point to an account you can't access, or to no account at all.